controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/SystemUpgrader.java

```diff
@@ -40,9 +40,12 @@ protected void upgrade(Version target, SystemApplication application, ZoneId zon
 
     @Override
     protected boolean convergedOn(Version target, SystemApplication application, ZoneId zone) {
-        return    minVersion(zone, application, Node::currentVersion).map(target::equals)
-                                                                     .orElse(true)
-               && application.configConvergedIn(zone, controller(), Optional.of(target));
+        Optional<Version> minVersion = minVersion(zone, application, Node::currentVersion);
+        // Skip application convergence check if there are no nodes belonging to the application in the zone
+        if (minVersion.isEmpty()) return true;
```

Comment:
Should add a test for this scenario, which I assume is "system without proxy app"?